src/whitenoise/accounts/relays.rs

```diff
@@ -154,152 +154,124 @@ impl Whitenoise {
-            account.nip65_relays.clone().into_iter().collect(),
-            RelayType::Nostr,
-        )
-        .await?;
-        self.disconnect_from_relay(account, relay).await
-    }
+        self.save_account(&account).await?;
 
-    pub async fn remove_inbox_relay(&self, account: &Account, relay: RelayUrl) -> Result<()> {
-        account.inbox_relays.remove(&relay);
-        self.update_account_relays_db(
-            &account.pubkey,
-            account.inbox_relays.clone().into_iter().collect(),
-            RelayType::Inbox,
-        )
-        .await?;
-        self.publish_relay_list_for_account(account, RelayType::Inbox)
+        // Update the in-memory account
+        {
+            let mut accounts = self.write_accounts().await;
+            accounts.insert(account.pubkey, account.clone());
+        }
+
+        // TODO: Do we need to manually disconnect or just wait for next session when we won't connect?
+
+        // We provide the prior relays to ensure we overwrite the relay list event on the relay we're leaving with the correct relays
+        self.publish_relay_list_for_account(&account, relay_type, &Some(current_relays))
             .await?;
-        self.disconnect_from_relay(account, relay).await
-    }
 
-    pub async fn remove_key_package_relay(&self, account: &Account, relay: RelayUrl) -> Result<()> {
-        account.key_package_relays.remove(&relay);
-        self.update_account_relays_db(
-            &account.pubkey,
-            account.key_package_relays.clone().into_iter().collect(),
-            RelayType::KeyPackage,
-        )
-        .await?;
-        self.publish_relay_list_for_account(account, RelayType::KeyPackage)
-            .await
+        Ok(())
     }
 
-    pub(crate) async fn publish_account_relay_info(&self, account: &Account) -> Result<()> {
+    pub(crate) async fn publish_account_relay_info(
+        &self,
+        account: &Account,
+        prior_relays: &Option<DashSet<RelayUrl>>,
+    ) -> Result<()> {
         if !self.logged_in(&account.pubkey).await {
             return Err(WhitenoiseError::AccountNotFound);
         }
 
-        self.publish_relay_list_for_account(account, RelayType::Nostr)
+        self.publish_relay_list_for_account(account, RelayType::Nostr, prior_relays)
             .await?;
-        self.publish_relay_list_for_account(account, RelayType::Inbox)
+        self.publish_relay_list_for_account(account, RelayType::Inbox, prior_relays)
             .await?;
-        self.publish_relay_list_for_account(account, RelayType::KeyPackage)
+        self.publish_relay_list_for_account(account, RelayType::KeyPackage, prior_relays)
             .await
     }
 
     pub(crate) async fn publish_relay_list_for_account(
         &self,
         account: &Account,
         relay_type: RelayType,
+        prior_relays: &Option<DashSet<RelayUrl>>, // If provided, this means at least one relay was removed. We need to publish to the prior relays as well.
```

Comment:
I can see how this got here as removing a relay is the use case that made this required, however I'd rename it so it's more descriptive of what it does. What do you think about something like 'relays_to_use', or 'target_relays' or  something like that? So basically we know that whatever we pass here means we're publishing to those relays.